### PR TITLE
Lorawan 1.0.4

### PR DESCRIPTION
**Pull request by mikev**

### CODE REVIEW

File: src/lora/lorawan_mac_region.erl [hunk 1]
```diff
@@ -211,7 +211,7 @@ max_payload_size(Region, DR) ->
         'US915' -> maps:get(DR, ?US915_PAYLOAD_SIZE_MAP, ?US915_MAX_DOWNLINK_SIZE);
         'AU915' -> maps:get(DR, ?AU915_PAYLOAD_SIZE_MAP, ?AU915_MAX_DOWNLINK_SIZE);
         'EU863' -> maps:get(DR, ?EU863_PAYLOAD_SIZE_MAP, ?EU863_MAX_DOWNLINK_SIZE);
-        _ -> maps:get(DR, ?US915_PAYLOAD_SIZE_MAP, ?US915_MAX_DOWNLINK_SIZE)
+        _ -> maps:get(DR, ?EU868_PAYLOAD_SIZE_MAP, ?EU868_MAX_DOWNLINK_SIZE)
```
Comment (lthiery) [inline src/lora/lorawan_mac_region.erl:214]: just curious, why EU868 as default?

Comment (mikev) [reply src/lora/lorawan_mac_region.erl:214]: The EU868 values are the most "generic".  Most other regions are just slight varations of EU868.  If we are missing the Region value, then this default should be the safest choice.

File: src/device/router_device_routing.erl [hunk 2]
```diff
@@ -528,6 +528,10 @@ validate_packet_offer(Offer, _Pid, Chain) ->
                             Error;
                         ok ->
                             PayloadSize = blockchain_state_channel_offer_v1:payload_size(Offer),
+                            Region = blockchain_state_channel_offer_v1:region(Offer),
+                            DR = 0, %% ToDo - Get SF and Bandwidth.  Use those values to calculate DR.  See - https://www.thethingsnetwork.org/docs/lorawan/modulation-data-rate/
```
[the remark this answers is not in the document]
Comment (mikev) [reply src/device/router_device_routing.erl:532]: This parrticular change is still a work-in-progress.

File: src/device/router_device.erl [hunk 3]
```diff
@@ -136,6 +137,18 @@ dev_nonces(Device) ->
...
+is_nonce_valid(DevNonce, Device) ->
+    %% For 1.1 and 1.0.4 end-devices LoRaWAN now specifies montonically incrementing nonce values.
+    %% Prior versions expected random value nonces.  Ideally we'd have a specific check for
+    %% devNonce > priorDevNonce.  For now keep the valid check consistent with random values.
+    case lists:member(DevNonce, Device:dev_nonces(Device)) of
+        true ->
+            false;
+        false ->
+            true
+    end.
```
Comment (xandkar) [inline src/device/router_device.erl:150]:
```suggestion
    not lists:member(DevNonce, Device:dev_nonces(Device)).
```

**Comment by macpie** [comment]
Closing this as we are moving the lora-lib

**Pull request closed without merging**